receiver/receivercreator/receiver.go

```diff
@@ -34,16 +34,27 @@ func newReceiverCreator(params receiver.Settings, cfg *Config) receiver.Metrics
 	}
 }
 
+// Host is an interface that the component.Host passed to receivercreator's Start function must implement
+type Host interface {
```

Comment:
does it need to be exported?

Reply:
I guess it's good to have the interface so that the component explicitly says what it needs from the `component.Host`. If it's controversial though, maybe we can make it private for now, enshrine it on some documentation on how to deal with this case, and make it public then?

Reply:
I can hide it for now.

Reply:
I am asking if it can be unexported because in general we are trying to keep the public API of any component down to its config struct if possible.

Do we see any code import this module and use this host interface?

Reply:
> Do we see any code import this module and use this host interface?

No, it doesn't need to be public at this time.